redbpf/src/lib.rs

```diff
@@ -392,6 +414,61 @@ impl Program {
     }
 }
 
+impl Drop for ProgramData {
+    fn drop(&mut self) {
+        if self.fd.is_some() {
+            unsafe {
+                let _ = libc::close(self.fd.unwrap());
+            }
+        }
+    }
+}
+
+fn pin_bpf_obj<P: AsRef<Path>>(fd: RawFd, pathname: P) -> Result<()> {
```

Comment:
You could also use the `pathname: impl AsRef<Path>` syntax here, that's a bit more modern.

Reply:
Oh I see. I didn't know which is modern or old style because I started learning rust with rust 2018. I just referred the code written in `std::path::Path` structure.
I'll reflect your suggestion.